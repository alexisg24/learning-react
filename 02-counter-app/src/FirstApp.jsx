import PropTypes from 'prop-types'

const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid='test-title'> {title} </h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: 'Your Hello World Here',
  subTitle: 'Your SubTitle Here!',
  name: 'Your Name Here!'
}

export { FirstApp }
