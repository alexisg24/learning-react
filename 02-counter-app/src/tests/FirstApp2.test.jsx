import { render, screen } from '@testing-library/react'
import { FirstApp } from '../FirstApp'

describe('Tests in <FirstApp />', () => {
  const title = 'Hola soy goku'
  const subtitle = 'Este es un subtitulo'

  test('Needs to match with snapshot', () => {
    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
  })

  test('Needs to show the exact title', () => {
    render(<FirstApp title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
  })

  test('Needs to show the title in h1', () => {
    render(<FirstApp title={title} />)
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
  })

  test('Needs to show the subtitle title used in props', () => {
    render(<FirstApp title={title} subTitle={subtitle} />)
    expect(screen.getAllByText(subtitle).length).toBe(2)
  })
})
