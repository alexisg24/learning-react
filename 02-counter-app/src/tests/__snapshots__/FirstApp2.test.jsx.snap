// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Needs to match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy goku
     
  </h1>
  <p>
    Your SubTitle Here!
  </p>
  <p>
    Your SubTitle Here!
  </p>
  <p>
    Your Name Here!
  </p>
</div>
`;
